homeassistant/components/tellduslive.py

```diff
@@ -360,3 +359,18 @@ def _last_updated(self):
         """Return the last update of a device."""
         return str(datetime.fromtimestamp(self.device.lastUpdated)) \
             if self.device.lastUpdated else None
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return "tellduslive-{}".format(self._id)
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            'connections': {(DOMAIN, self.device.client)},
```

Comment:
`DOMAIN` is not a valid entry for `connections`.

If `self.device.client` is unique, how is it different from `self.unique_id` ?

Reply:
A client in tellduslive thermology is a gateway by the homeassistant thermology.

Reply:
Ah, it should be `via_hub`. Thank you for spotting this.

Reply:
The entities are connected to the "hub" via this connection right?

So I should add `via_hub` too?

Reply:
You should drop connections unless you have a valid connections specifier. 

For via_hub, you need to specify the device that is used as gateway but that gateway also needs to be registered, as per the docs.

Reply:
If I'm not mistaken is `device_info` just used if we are using **Config Entries**, should I remove it now and readd it when it is time for a **Config Flow**?

Reply:
Yes and yes, good point.

Reply:
> A client in tellduslive thermology is a gateway by the homeassistant thermology.

@fredrike, feel free to rename this if it makes it clearer.

Sorry for the "id"-confusion. The key is on this line, I think: https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/tellduslive.py#L264 where a sensor is identified both by the device id (meaning the physical device measuring stuff and sending it to the cloud), and the quantity (e.g. temperature or humidity) as well as the scale (some devices report values in more than one scale, e.g. percent vs mm, C vs F etc). 

Btw thanks for looking into this! I have myself moved my setup from the Telldus live cloud solution to using MQTT for all my devices and sensors (since the cloud solution doesn't support instant "push" notifications anyway) so I haven't given this component any attention lately.

Reply:
@molobrakos It is tellduslive that calls it client.

Reply:
> Yes and yes, good point.

I've removed that part of the code in ea8043a, I'll take a look at implementing **Config Flow** later :).